tools/redhat/osv.py

```diff
@@ -0,0 +1,195 @@
+    # This class encapsulates and validates version range events
+    # Only a single public method is required for custom JSON encoding
+
+    INTRODUCED = "introduced"
+    FIXED = "fixed"
+
+    def __init__(self, event_type: str, version: str = "0"):
+        expected = (self.INTRODUCED, self.FIXED)
+        if event_type not in expected:
+            raise ValueError(f"Expected one of {expected} for type. Got {event_type}")
+        self.event_type = event_type
+        self.version = version
+
+    def encode_json(self):
+        """
+        Custom JSON encoding for event type which changes attribute name depending on the type of
+        event eg. introduced or fixed
+        """
+        if self.event_type == Event.INTRODUCED:
+            return {Event.INTRODUCED: self.version}
+        if self.event_type == Event.FIXED:
+            return {Event.FIXED: self.version}
+        raise ValueError("Unexpected event_type for Event")
+
+
+class Range:
+    """
+    Class to hold range information for a Package. Ecosystem here refers to RPM versions as defined
+    in https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c
+    """
+
+    # pylint: disable=too-few-public-methods
+    # This class encapsulates version range types as 'ECOSYSTEM' type
+    # Only initialization is required because data retrieval is via JSON encoding
+
+    def __init__(self, fixed: str):
+        self.type = "ECOSYSTEM"
+        self.events = [Event("introduced"), Event("fixed", fixed)]
+
+
+class Package:
+    """
+    Class to hold package data for an Affect. Expects an ecosystem string that starts with
+    CPE_PATTERN.
+    Replaces the CPE prefix 'redhat' part with 'Red Hat' to match more closely with other ecosystem
+    identifiers in the OSV database
+    """
+
+    # pylint: disable=too-few-public-methods
+    # This class encapsulates Red Hat RPM Packages by Ecosystem
+    # Only initialization is required because data retrieval is via JSON encoding
+
+    CPE_PATTERN = re.compile(r"cpe:/[oa]:(redhat)")
+
+    def __init__(self, name: str, ecosystem: str, purl: str):
+        self.name = name
+        if not self.CPE_PATTERN.match(ecosystem):
+            raise ValueError(f"Got unsupported ecosystem: {ecosystem}")
+        self.ecosystem = f"Red Hat{self.CPE_PATTERN.split(ecosystem, maxsplit=1)[-1]}"
+        self.purl = purl
+
+
+class Affected:
+    """
+    Class to hold affected data for a Vulnerability
+    """
+
+    # pylint: disable=too-few-public-methods
+    # This class encapsulates Red Hat Affects
+    # Only initialization is required because data retrieval is via JSON encoding
+
+    def __init__(self, remediation: Remediation):
+        self.package = Package(remediation.component, remediation.cpe, remediation.purl)
+        self.ranges = [Range(remediation.fixed_version)]
+
+
+class OSV:
+    """
+    Class to convert CSAF data to OSV
+    """
+
+    SCHEMA_VERSION = "1.6.4"
```

Comment:
Update to currently released version

```suggestion
    SCHEMA_VERSION = "1.6.5"
```